File: CS/Solution28.Module.Web/Controllers/ViewControllerWithParametrizedAction.cs
using System;
using DevExpress.Data.Filtering;
using DevExpress.ExpressApp;
using DevExpress.ExpressApp.Actions;
using DevExpress.ExpressApp.Web.SystemModule;

namespace Solution28.Module.Web {
    public class ViewController1 : ViewController {
        public ViewController1() {
            ParametrizedAction filterAction = new ParametrizedAction(this, "FilterRange", "Edit", typeof(DateTime));
            filterAction.Execute += new ParametrizedActionExecuteEventHandler(filterAction_Execute);
        }
        private void filterAction_Execute(object sender, ParametrizedActionExecuteEventArgs e) {
            Range<DateTime> range = (Range<DateTime>)e.ParameterCurrentValue;
            ((ListView)View).CollectionSource.Criteria["By Date"] = CriteriaOperator.Parse("[Date] <= ? AND [Date] >= ?", range.To, range.From);
        }
    }    
}

File: VB/Solution28.Module.Web/Controllers/ViewControllerWithParametrizedAction.vb
Imports System
Imports DevExpress.Data.Filtering
Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Actions
Imports DevExpress.ExpressApp.Web.SystemModule

Namespace Solution28.Module.Web
    Public Class ViewController1
        Inherits ViewController

        Public Sub New()
            Dim filterAction As New ParametrizedAction(Me, "FilterRange", "Edit", GetType(Date))
            AddHandler filterAction.Execute, AddressOf filterAction_Execute
        End Sub
        Private Sub filterAction_Execute(ByVal sender As Object, ByVal e As ParametrizedActionExecuteEventArgs)
            Dim range As Range(Of Date) = CType(e.ParameterCurrentValue, Range(Of Date))
            CType(View, ListView).CollectionSource.Criteria("By Date") = CriteriaOperator.Parse("[Date] <= ? AND [Date] >= ?", range.To, range.From)
        End Sub
    End Class
End Namespace
